test/onnx/gen_onnx.py

```diff
@@ -5790,6 +5790,132 @@ def slice_test():
     return ([node], [x], [y])
 
 
+@onnx_test
+def slice_dyn_test():
+    x = helper.make_tensor_value_info('0', TensorProto.FLOAT, [None, 2])
+    y = helper.make_tensor_value_info('1', TensorProto.FLOAT, [None, 2])
+
+    node = onnx.helper.make_node('Slice',
+                                 inputs=['0'],
+                                 axes=[0, 1],
+                                 starts=[1, 0],
+                                 ends=[2, 2],
+                                 outputs=['1'])
+
+    return ([node], [x], [y])
+
+
+@onnx_test
+def slice_step_dyn_test():
```

Comment:
Since this is only used inside onnx_test, you could have just kept using literal values (e.g. as in slice_dyn_test). instead of making `nodes` for `axes`, `steps`, `end` etc. It doesn't hurt though.  You can keep this suggestion for future to minimize code changes. No need to change for this PR.